config/hotreloadable.go

```diff
@@ -1,14 +1,64 @@
 package config
 
-import "time"
+import (
+	"strings"
+	"sync"
+	"time"
+)
 
 // RegisterIntConfigVariable registers int config variable
-func RegisterIntConfigVariable(defaultValue int, ptr *int, isHotReloadable bool, valueScale int, keys ...string) {
-	Default.RegisterIntConfigVariable(defaultValue, ptr, isHotReloadable, valueScale, keys...)
+// WARNING: a different order of keys will result in a different variable
+// Deprecated: use RegisterIntVar or RegisterReloadableIntVar instead
+func RegisterIntConfigVariable(defaultValue int, ptr *int, isHotReloadable bool, valueScale int, orderedKeys ...string) {
+	Default.RegisterIntConfigVariable(defaultValue, ptr, isHotReloadable, valueScale, orderedKeys...)
+}
+
+// RegisterIntVar registers a not hot-reloadable int config variable
+// WARNING: a different order of keys will result in a different variable
+func RegisterIntVar(defaultValue, valueScale int, orderedKeys ...string) *int {
+	return Default.RegisterIntVar(defaultValue, valueScale, orderedKeys...)
+}
+
+// RegisterReloadableIntVar registers a hot-reloadable int config variable
+// WARNING: a different order of keys will result in a different variable
+func RegisterReloadableIntVar(defaultValue, valueScale int, orderedKeys ...string) *Reloadable[int] {
+	return Default.RegisterReloadableIntVar(defaultValue, valueScale, orderedKeys...)
 }
 
 // RegisterIntConfigVariable registers int config variable
-func (c *Config) RegisterIntConfigVariable(defaultValue int, ptr *int, isHotReloadable bool, valueScale int, keys ...string) {
+// WARNING: a different order of keys will result in a different variable
+// Deprecated: use RegisterIntVar or RegisterReloadableIntVar instead
+func (c *Config) RegisterIntConfigVariable(
+	defaultValue int, ptr *int, isHotReloadable bool, valueScale int, orderedKeys ...string,
+) {
+	c.registerIntVar(defaultValue, ptr, isHotReloadable, valueScale, func(v int) {
+		*ptr = v
+	}, orderedKeys...)
+}
+
+// RegisterIntVar registers a not hot-reloadable int config variable
+// WARNING: a different order of keys will result in a different variable
+func (c *Config) RegisterIntVar(defaultValue, valueScale int, orderedKeys ...string) *int {
+	ptr := getOrCreatePointer(c.vars, c.varsMisuses, &c.varsLock, defaultValue, false, orderedKeys...).(*int)
```

Comment:
We don't really want to cache anything for non-reloadable configuration options, do we? Non-reloadable config option requests can simply be directly delegated to the underlying config library (viper)